Code/Tools/FBuild/FBuildCore/Graph/ObjectListNode.cpp

```diff
@@ -353,7 +360,15 @@ bool ObjectListNode::CreateDynamicObjectNode( Node * inputFile, const DirectoryL
             // ... use everything after that
             subPath.Assign(fileName.Get() + baseDir.GetLength(), lastSlash ); // includes last slash
         }
-    }
+	}
+	else
+	{
+		if ( !m_BaseDirectory.IsEmpty() && fileName.BeginsWith( m_BaseDirectory ) )
```

Comment:
Need to do this:
```C++
if ( !m_BaseDirectory.IsEmpty() && PathUtils::PathBeginsWith( fileName, m_BaseDirectory ) )
```
to correctly handle case-sensitivity on all platforms. (Currently this is incorrectly case sensitive on Windows and OS X)